javascript/index.js

```diff
@@ -4,37 +4,175 @@
+              "mashedPotatoes",
+              3,
+              (step3) => {
+                document.querySelector(
+                  "#mashedPotatoes"
+                ).innerHTML += `<li>${step3}</li>`;
+                getInstruction(
+                  "mashedPotatoes",
+                  4,
+                  (step4) => {
+                    document.querySelector(
+                      "#mashedPotatoes"
+                    ).innerHTML += `<li>${step4}</li>`;
+                    getInstruction(
+                      "mashedPotatoes",
+                      5,
+                      (step5) => {
+                        document.querySelector(
+                          "#mashedPotatoes"
+                        ).innerHTML += `<li>${step5}</li>`;
+                        getInstruction(
+                          "mashedPotatoes",
+                          6,
+                          () => {
+                            document.querySelector(
+                              "#mashedPotatoes"
+                            ).innerHTML += `<li>${"Mashed potatoes are ready!"}</li>`;
+                          },
+                          (error) => console.log(error)
+                        );
+                      },
+                      (error) => console.log(error)
+                    );
+                  },
+                  (error) => console.log(error)
+                );
+              },
+              (error) => console.log(error)
+            );
+          },
+          (error) => console.log(error)
+        );
+      },
+      (error) => console.log(error)
+    );
+  },
+  (error) => console.log(error)
+);
 
 // Iteration 2 - using promises
-// ...
+obtainInstruction("steak", 0)
+  .then((step0) => {
+    document.querySelector("#steak").innerHTML += `<li>${step0}</li>`;
+    return obtainInstruction("steak", 1);
+  })
+  .then((step1) => {
+    document.querySelector("#steak").innerHTML += `<li>${step1}</li>`;
+    return obtainInstruction("steak", 2);
+  })
+  .then((step2) => {
+    document.querySelector("#steak").innerHTML += `<li>${step2}</li>`;
+    return obtainInstruction("steak", 3);
+  })
+  .then((step3) => {
+    document.querySelector("#steak").innerHTML += `<li>${step3}</li>`;
+    return obtainInstruction("steak", 4);
+  })
+  .then((step4) => {
+    document.querySelector("#steak").innerHTML += `<li>${step4}</li>`;
+    return obtainInstruction("steak", 5);
+  })
+  .then((step5) => {
+    document.querySelector("#steak").innerHTML += `<li>${step5}</li>`;
+    return obtainInstruction("steak", 6);
+  })
+  .then((step6) => {
+    document.querySelector("#steak").innerHTML += `<li>${step6}</li>`;
+    return obtainInstruction("steak", 7);
+  })
+  .then((step7) => {
+    document.querySelector("#steak").innerHTML += `<li>${step7}</li>`;
+    return obtainInstruction("steak", 8);
+  })
+  .then(() => {
+    document.querySelector(
+      "#steak"
+    ).innerHTML += `<li>${"Steak is ready!"}</li>`;
```

Comment:
same here and you don't need to use template literals, you can just write the sentence or you declare a variable and save the sentence before and after that use the variable names with template literals

```
 .then((step5) => {
      document.querySelector("#steak").innerHTML += `<li>${step5}</li>`;
      return obtainInstruction("steak", 6);
    })
    .then((step6) => {
        document.querySelector("#steak").innerHTML += `<li>${step6}</li>`;
        return obtainInstruction("steak", 7);
      })
    .then((step7) => {
      document.querySelector("#steak").innerHTML += `<li>${step7}</li>`;
      document.querySelector( "#steak").innerHTML += `<li>Steak is ready!</li>`;
      document.querySelector("#steakImg").removeAttribute("hidden");
    });

```